### PR TITLE
Kamu UI 464 query explainer successfully validates commitment but shows no data

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**
Closes: https://github.com/kamu-data/kamu-web-ui/issues/464

image

### CODE REVIEW

File: src/app/query-explainer/components/input-data-section/input-data-section.component.ts [hunk 1]
```diff
@@ -23,18 +25,47 @@ export class InputDataSectionComponent {
...
+        switch (option) {
             case "ArrowJson":
                 return "Arrow JSON";
+            case "Parquet":
+                return "Parquet";
+            case "ParquetJson":
+                return "Parquet JSON";
+            /* istanbul ignore next */
+            default:
+                return "Unknown schema format";
```
Comment (sergiimk) [inline src/app/query-explainer/components/input-data-section/input-data-section.component.ts:54]: same here

File: src/app/query-explainer/components/reproduced-result-section/reproduced-result-section.component.ts [hunk 2]
```diff
@@ -10,20 +10,59 @@ import { DataRow, DataSchemaField } from "src/app/interface/dataset.interface";
...
+            }
+            case "JsonAoA": {
+                return this.parseDataFromJsonAoAFormat(output.data, columnNames);
+            }
+            case "JsonSoA": {
+                return this.parseDataFromJsonSoAFormat(output.data, columnNames);
+            }
+            /* istanbul ignore next */
+            default:
+                throw new Error("Unable to parse date");
```
Comment (sergiimk) [inline src/app/query-explainer/components/reproduced-result-section/reproduced-result-section.component.ts:29]: As discussed - UI should modify `input.dataFormat` to request data in a format that suits it best.

File: src/app/query-explainer/query-explainer.service.spec.ts [hunk 3]
```diff
@@ -46,7 +46,7 @@ describe("QueryExplainerService", () => {
         const errorMessage = "Error";
         const toastrServiceSpy = spyOn(toastrService, "error");
 
-        service.processQuery(mockQuery).subscribe({
+        service.processQuery(mockQuery, ["Proof"]).subscribe({
```
Comment (sergiimk) [inline src/app/query-explainer/query-explainer.service.spec.ts:49]: Make sure that the request for data is not asking for `Proof` - we just need the data and hashing + signing to create the proof is computationally expensive.

File: src/app/query-explainer/components/input-data-section/input-data-section.component.ts [hunk 4]
```diff
@@ -23,18 +25,47 @@ export class InputDataSectionComponent {
...
+            case "JsonAoS":
+                return "JSON AoS";
+            case "JsonSoA":
+                return "JSON SoA";
+            case "JsonAoA":
+                return "JSON AoA";
             /* istanbul ignore next */
             default:
-                return "Unknown options";
+                return "Unknown data format";
```
[the remark this answers is not in the document]
Comment (dmitriy-borzenko) [reply src/app/query-explainer/components/input-data-section/input-data-section.component.ts:68]: Fixed, remove data format mapper

File: src/app/query-explainer/components/input-data-section/input-data-section.component.ts (hunk 1 above)
Comment (dmitriy-borzenko) [reply src/app/query-explainer/components/input-data-section/input-data-section.component.ts:54]: Fixed, removed schema format mapper

File: src/app/query-explainer/components/reproduced-result-section/reproduced-result-section.component.ts (hunk 2 above)
Comment (dmitriy-borzenko) [reply src/app/query-explainer/components/reproduced-result-section/reproduced-result-section.component.ts]: Removed this code

File: src/app/query-explainer/query-explainer.service.spec.ts (hunk 3 above)
Comment (dmitriy-borzenko) [reply src/app/query-explainer/query-explainer.service.spec.ts:49]: Fixed

File: src/app/query-explainer/query-explainer.service.ts [hunk 5]
```diff
@@ -13,17 +18,47 @@ export class QueryExplainerService {
...
+            map((response: QueryExplainerResponse) => {
+                const cloneData = Object.assign({}, response);
+                if ("output" in cloneData) {
+                    delete cloneData.output;
+                }
+                return cloneData as QueryExplainerProofResponse;
+            }),
+            tap((res: QueryExplainerProofResponse) => {
+                if (res.subQueries?.length) {
+                    throw new Error("Unknown subQueries array");
```
Comment (zaychenko-sergei) [inline src/app/query-explainer/query-explainer.service.ts:43]: Unsupported

File: src/app/query-explainer/query-explainer.component.ts [hunk 6]
```diff
@@ -86,4 +100,30 @@ export class QueryExplainerComponent extends BaseComponent implements OnInit {
...
+                map(([commitment, dataJsonAoS, sqlQueryVerify]) => {
+                    return {
+                        sqlQueryExplainerResponse: {
+                            ...commitment,
+                            ...dataJsonAoS,
+                        },
+                        sqlQueryVerify,
+                    };
+                }),
+            );
```
Comment (zaychenko-sergei) [inline src/app/query-explainer/query-explainer.component.ts:124]: The code looks duplicated

File: src/app/query-explainer/query-explainer.component.ts [hunk 7]
```diff
@@ -34,26 +43,31 @@ export class QueryExplainerComponent extends BaseComponent implements OnInit {
...
+                            };
+                        }),
                     );
                 }),
             );
         }
     }
 
-    private commitmentDataWithoutOutput(commitmentUploadToken: string): Observable<QueryExplainerResponse> {
+    private uploadCommitment(commitmentUploadToken: string): Observable<QueryExplainerResponse> {
```
Comment (zaychenko-sergei) [inline src/app/query-explainer/query-explainer.component.ts:70]: Seems like it should be "download", not "upload".

File: src/app/project-links.ts [hunk 8]
```diff
@@ -13,7 +13,7 @@ export default class ProjectLinks {
     public static readonly URL_SETTINGS: string = "v/settings";
     public static readonly URL_ADMIN_DASHBOARD: string = "v/admin-dashboard";
     public static readonly URL_FLOW_DETAILS: string = "flow-details";
```
Comment (zaychenko-sergei) [inline src/app/project-links.ts:15]: I guess should be `/v/flow-details` too

Comment (dmitriy-borzenko) [reply src/app/project-links.ts:15]: Fixed

File: src/app/query-explainer/query-explainer.component.ts (hunk 7 above)
Comment (dmitriy-borzenko) [reply src/app/query-explainer/query-explainer.component.ts:70]: Cahnged

File: src/app/query-explainer/query-explainer.component.ts (hunk 6 above)
Comment (dmitriy-borzenko) [reply src/app/query-explainer/query-explainer.component.ts:124]: Moved to separated method duplicated code

File: src/app/project-links.ts (hunk 8 above)
Comment (zaychenko-sergei) [reply src/app/project-links.ts:15]: Come one, then why the hell it is declared there among with other urls, as if it's from the same group...

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**